src/core/psxmem.cc

```diff
@@ -92,35 +92,43 @@ int PCSX::Memory::init() {
     m_writeLUT = (uint8_t **)calloc(0x10000, sizeof(void *));
 
     m_psxM = (uint8_t *)calloc(0x00800000, 1);
-    m_psxP = (uint8_t *)calloc(0x00010000, 1);
+    m_psxP = (uint8_t *)calloc(0x00080000, 1);
     m_psxH = (uint8_t *)calloc(0x00010000, 1);
     m_psxR = (uint8_t *)calloc(0x00080000, 1);
 
-    if (m_readLUT == NULL || m_writeLUT == NULL || m_psxM == NULL || m_psxP == NULL || m_psxH == NULL) {
+    if (m_readLUT == NULL || m_writeLUT == NULL || m_psxM == NULL || m_psxP == NULL || m_psxR == NULL || m_psxH == NULL) {
         g_system->message("%s", _("Error allocating memory!"));
         return -1;
     }
 
-    // MemR
-    m_readLUT[0x1f00] = (uint8_t *)m_psxP;
+    // MemR / MemW
```

Comment:
It's also probably best to do away with the old code's tendency of having comments for self-explanatory code written in half-legible English. Not much necessity in specifying "MemR/MemW" when it's obviously made R/W below